logstash-core/src/main/java/org/logstash/instrument/witness/PipelinesWitness.java

```diff
@@ -27,4 +36,46 @@ public PipelineWitness pipeline(String name) {
         return pipelines.computeIfAbsent(name, k -> new PipelineWitness(k));
     }
 
+    @Override
+    public void genJson(JsonGenerator gen, SerializerProvider provider) throws IOException {
+        new Serializer().innerSerialize(this, gen, provider);
+    }
+
+    /**
+     * The Jackson serializer.
+     */
+    public static class Serializer extends StdSerializer<PipelinesWitness> {
+
+        /**
+         * Default constructor - required for Jackson
+         */
+        public Serializer() {
+            this(PipelinesWitness.class);
+        }
+
+        /**
+         * Constructor
+         *
+         * @param t the type to serialize
+         */
+        protected Serializer(Class<PipelinesWitness> t) {
+            super(t);
+        }
+
+        @Override
+        public void serialize(PipelinesWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeStartObject();
+            innerSerialize(witness, gen, provider);
+            gen.writeEndObject();
+        }
+
+        void innerSerialize(PipelinesWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeObjectFieldStart(KEY);
+            for (Map.Entry<String, PipelineWitness> entry : witness.pipelines.entrySet()) {
```

Comment:
Same as in other spots, since we have a serializer for `PipelineWitness` we should be able to simply write the whole `Map` to Jackson via `com.fasterxml.jackson.core.JsonGenerator#writeObject`?